File: backend/src/main/java/com/lawencon/ticketms/service/RoleService.java
package com.lawencon.ticketms.service;

import com.lawencon.ticketms.dto.role.RoleListResDto;
import com.lawencon.ticketms.dto.role.RoleResDto;

public interface RoleService {
	RoleResDto getById(Long id);
	RoleListResDto getAll();
}

File: frontend/src/app/service/role.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_URL } from "../constant/base.url";
import { RoleListResDto } from "../dto/role/role.list.res.dto";

@Injectable({
    providedIn: 'root'
})
export class RoleService {
    constructor(private http: HttpClient) { }

    getAll() : Observable<RoleListResDto>{
        return this.http.get<RoleListResDto>(`${BASE_URL.LOCALHOST}/roles`);
    }
}